service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -217,6 +221,7 @@ public DatasetId createDataset(
       StorageSystem storageSystem,
       String storageSourceId,
       String metadata) {
+    validateMetadata(metadata);
     var dataset = new Dataset(storageSourceId, storageSystem, metadata);
```

Comment:
It occurs to me that both validation and conversion from String to ObjectNode are really `Dataset` operations. Dataset itself should disallow creation with invalid JSON metadata and support a way to get the content as `ObjectNode`. But I don't see a way to do that without making our `ObjectMapper` instance a global variable.